book_source/adv_user_guide_web/Benchmarking.Rmd

```diff
@@ -0,0 +1,84 @@
+
+3. A format record (Instructions [here](#NewFormat)))
+
+
+## Benchmarking in pecan.xml
+
+The `pecan.xml` has an _optional_ benchmarking section. Ultimately, the user should not have to edit this document by hand, but it is still necessary for cloning runs. 
+
+Below are all the possible benchmarking sections. 
+If you provide the ensemble id, then you do not need to provide the reference run id and vice versa.
+
+`<benchmarking>` settings:
+
+- `ensemble_id`: the id of the ensemble that you will be using - the settings from this ensemble will be saved in a reference run record and then `ensemble_id` will be replaced with `reference_run_id` 
+- `input_id`: the id(s) of the benchmarking data  (required)
+- `variable_id`: the id(s) of the variable(s) of interest within the data. If you leave this blank, all variables that are shared between the input and model output will be used. 
+- `metric_id`: the id(s) of the metric(s) to be calculated. If you leave this blank, all metrics will be used. 
+- `new_run`: TRUE = create new run, FALSE = use existing run (required)
+
+Example:
+
+```xml
+ <benchmark>
+     <ensemble_id>1000004796</ensemble_id> 
+     <input_id>1000008121</input_id>
+     <variables>
+       <variable_id>411</variable_id> 
+     </variables>
+     <metrics>
+       <metric_id>1000000003</metric_id> 
+     </metrics>
+     <new_run>TRUE</new_run>
+   <reference_run_id>1000000050</reference_run_id>
+ </benchmark>
+```
+
+## 1. New Run
+
+- Setup your run (preferably using the pecan interface). 
+- When the run has completed, navigate to the workflow visualization Shiny app. 
+- Load model data
+  + Make sure that your model output is loading properly (i.e. you can see plots of your data)
+- Load benchmarking data
+  + Again make sure that you can see the uploaded data plotted alongside the model output. In the future there will be more tools for double checking that your uploaded data is appropriate for benchmarking, but for now you need to do the sanity checks by  hand. 
+- Navigate to the Benchmarking tab
+  + The first step is to register the new model run as a reference run in the database. Benchmarking cannot be done before this step is completed. When the reference run record has been created, additional menus for benchmarking will appear. 
+  + Next from the menu select
+    + The variables in the uploaded data that you wish to compare with model output. Note: this list shows all available variables regardless of whether or not there is a comparable variable in the model output. Thus if you select a variable to be benchmarked and it does not show up in the results, double check that it is also in the model output! 
+    + The numerical metrics you would like to use in your comparison. 
+    + Additional comparison plots that you would like to see.
+  + All these selections populate the benchmarking section of the xml witch is then saved in the same location as the original run output. 
```

Comment:
which

Also, writing this out to an xml just so you can read back in seems unnecessary and overly complicated. Can't you just call the required R function directly, passing it in the required variables that you go from the pull-down menus?